server/etcdserver/api/membership/cluster.go

```diff
@@ -530,8 +531,13 @@ func (c *RaftCluster) PromoteMember(id types.ID, shouldApplyV3 ShouldApplyV3) {
 	if c.v2store != nil {
 		mustUpdateMemberInStore(c.lg, c.v2store, c.members[id])
 	}
-	if c.be != nil && shouldApplyV3 {
-		unsafeSaveMemberToBackend(c.lg, c.be, c.members[id])
+	if c.be != nil {
+		if shouldApplyV3 {
+			unsafeSaveMemberToBackend(c.lg, c.be, c.members[id])
+		} else {
+			c.lg.Info("Forcibly apply member promotion", zap.String("member", fmt.Sprintf("%+v", *c.members[id])))
```

Comment:
Could we only force promotion when it's needed? If `shouldApplyV3` is false and `Member.Learner` is true, means this member was not promoted in v3 and we fixing it here. Also please leave a comment explaining it.